### PR TITLE
BRAWL - Unarmed Parity/Bugfixes

### PR DESCRIPTION
**Pull request by kiwedespars**
## About The Pull Request
UNARMED PARITY.

```
- GENERAL -

- Punches now reduce blade sharpness.
- Punch misscost removed.
- Katars clickdelay nerfed to 10, and also have their 1.3 damage intent reduced to 1.2 damage.
```
```
- INTENTS -

- !! BOTH STRONG AND SWIFTCOST MOVED UP FROM 1.5 TO 6 COST !!
- Strong punches now deal 15% extra damage properly.
- Strong punches now deal extra sharpness and integrity damage
- Grab intent cost reduced from 10 to 2, as to account for the swift/strongcost being fiddled with.
```
```
- GUARDING/RIPOSTE/CLASHING -

- Unarmed now fails a riposte for trying to swing while performing one.
- Unarmed riposte now properly does sharpness and integrity damage.
- Unarmed can now clash. Winning a clash against someone unarmed makes them vulnerable and incurs clickdelay, since they can't be disarmed.
- Riposting unarmed now gives them exposed and clickdelay, similar to using it against a weapon.
```
```
- BUGFIX -

- Punches now give dodge back when being hit without armor to mitigate it.
- Punches now pause skinarmor regeneration properly
- Bracers, knuckles, and armwraps now take damage properly when parrying with unarmed
- Unarmed no longer gives you an effective -6 INT malus when being feinted.
- Stomps have 100% accuracy again, because they're stomps.
- Skilldifference actually matters now for clashing.
```

## Testing Evidence

image

image

## Why It's Good For The Game

```
- GENERAL -
```
Punches not interacting with the sharpness system just seemed off/strange while I was fixing the fact it has `no` benefits from strong intent. It doesn't do an absurd amount of damage to warrant it, being more-or-less on par with the sabre/rapier.

Punch misscosts removed because nothing else has a misscost outside of charged intents, to my knowledge.

Katars get a bit of a nerf because people pointed out that they're essentially super-sabres with 2x sharpness reduction. This should bring them in line a little bit more, keeping in mind they do just chip the moment they get riposted.

Riposting is pretty huge, being able to stem the tide of blows for three seconds and also get a 2.5x damage hit to their armor is incredibly important.
```
- SWIFTSTAM COST -
```
Moving the cost from 1.5 to 6 should solve the feeling of unarmed doing insane damage per second, because their clickdelay was actually super low for essentially extra zero cost (only .5 more). Killing the hyper-efficiency of it should be healthier for the game. 

As for why it's `6` and not `10` as to bring it to true parity, is because punching does the least amount of damage per hit (Assuming 14 strength, civbarb, and knuckles which sacrifice protections for +8 damage, 36 damage) and were, when swifting alongside a grand mace and a longsword, the slowest weapon to break through a single layer of plate. (bucket helmet, 400 integrity)

The grand mace would break it in `6` hits - `60` stam, `4.2 seconds`, the longsword in `9` hits - `90` stam, `6.03 seconds` and fists in `13` hits, `78` stamina, `7.3 seconds`. This gives them their niche of being slightly stamina-efficient and keeps in mind the fact they're the lowest damage per hit, and thats assuming it on the best armor to hit into.

ideally all this puts it into a better spot both to use and be against

## Changelog

:cl:
fix: unarmed now properly lowers sharpness
fix: strong intent now benefits unarmed
fix: unarmed fails riposte if they punch during it
fix: punches now give dodge back on hits to skin
fix: fixes unarmed not pausing skinarmor regen
fix: feint cares about unarmed now (no free +60% to being feinted)
fix: bracers/knuckles now take damage properly when parrying with unarmed
balance: stomps have 100% accuracy again
balance: swift/strongcost for unarmed 1.5 -> 6
balance: misscost for unarmed 3 -> 1
balance: katar cut 1.3 -> 1.2. katar clickdelay for both intents 8 -> 10
balance: grab cost reduced to account for swiftcost adjustment
balance: riposting unarmed grants normal riposte counter
/:cl:

### CODE REVIEW

**Comment by magnitude9girthquake** [comment]
Do you think you could also fix the unarmed dodge chances, too? As it stands right now, unarmed hits don't restore any dodge percentage

**Comment by kiwedespars** [comment]
image

**Comment by FreeStylaLT** [comment]
If you're wishing to standardise punching across the board this would have to include punches working on riposte as well.

**Comment by SnubbyNotSorry** [comment]
Oh thanks a LOT for the strong intent one, I had noticed it before !

**Comment by LoJoFan** [comment]
Given how this fixes unarmed not damaging the defender's weapon sharpness and durability, I think this PR should also try to address...
- Unarmed parries not damaging bracer slot/glove slot
- Unarmed swift punch not draining as much stamina as other weapon swift strikes would
Especially the second point is worrisome: fighting SWIFT punch is already annoying due to the little stam drain the attacker takes; with this change, using DEFEND intent to deal with SWIFT punch will just nuke your weapon and eventually leave you defenseless, with some punches still going through and damaging you (due to a larger volume, duh), and at a likely stamina/energy disparity.

**Comment by kiwedespars** [comment]
`Unarmed parries not damaging bracer slot/glove slot`

fixed

`Unarmed swift punch not draining as much stamina as other weapon swift strikes would`

adjusted to be 2x what it currently is, from 1.5 to 3. might still be too low, but keep in mind that despite the volume of punches, they still break through armor-integrity the slowest out of all weapons on swift.

edit: 1.5 -> 8 to bring it up to parity just to see, probably ready for tm

File: code/modules/mob/living/combat/azure_combat.dm [hunk 1]
```diff
@@ -111,20 +117,20 @@
...
 	else
-		instantloss = TRUE	//We are Guarding with a book or something -- no chance for us.
+		skilldiff = get_skill_level(/datum/skill/combat/unarmed)
 
-	if(IU.associated_skill)
+	if(IU?.associated_skill)
 		skilldiff = skilldiff - HU.get_skill_level(IU.associated_skill)
 	else
-		instantwin = TRUE	//THEY are Guarding with a book or something -- no chance for them.
+		skilldiff = HU.get_skill_level(/datum/skill/combat/unarmed)
```
Comment (WeNeedMorePhoron) [inline code/modules/mob/living/combat/azure_combat.dm]: Should be skilldiff = skilldiff - HU.get_skill_level(/datum/skill/combat/unarmed), otherwise the ATTACKER's skills help them

File: code/modules/mob/living/combat/azure_combat.dm [hunk 2]
```diff
@@ -5,36 +5,44 @@
...
 	var/mob/living/carbon/human/H = user
-	if(!IU)	//The opponent is trying to rawdog us with their bare hands while we have Guard up. We get a free attack on their active hand.
-		if(!IM)	//We are also unarmed -- no clash or riposte without a weapon on the guarder's side.
-			remove_status_effect(/datum/status_effect/buff/clash)
-			return
-		var/obj/item/bodypart/affecting = H.get_bodypart("[(user.active_hand_index % 2 == 0) ? "r" : "l" ]_arm")
-		var/force = get_complex_damage(IM, src)
-		var/armor_block = H.run_armor_check(BODY_ZONE_PRECISE_L_HAND, used_intent.item_d_type, armor_penetration = used_intent.penfactor, damage = force, used_weapon = IM)
-		if(H.apply_damage(force, IM.damtype, affecting, armor_block))
-			visible_message(span_suicide("[src] gores [user]'s hands with \the [IM]!"))
```
Comment (WeNeedMorePhoron) [inline code/modules/mob/living/combat/azure_combat.dm:16]: You removed the goring. Is this intended? If so, document it

**Review by WeNeedMorePhoron — APPROVED** [review]
Most looks good. One big concern before I can TM this

File: code/modules/mob/living/combat/azure_combat.dm (hunk 1 above)
Comment (kiwedespars) [reply code/modules/mob/living/combat/azure_combat.dm]: whoops

File: code/modules/mob/living/combat/azure_combat.dm (hunk 2 above)
Comment (kiwedespars) [reply code/modules/mob/living/combat/azure_combat.dm:16]: ye it's uhh

balance: riposting unarmed grants normal riposte counter
